airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -112,6 +112,9 @@ def get_json_schema(self) -> Mapping[str, Any]:
     def as_airbyte_stream(self) -> AirbyteStream:
         stream = AirbyteStream(name=self.name, json_schema=dict(self.get_json_schema()), supported_sync_modes=[SyncMode.full_refresh])
 
+        if self.namespace:
```

Comment:
This should be declared as an optional property on stream rather than on abstract source, because namespace can be different per each stream. It should look exactly the same as cursor field (minus the type difference)

Reply:
I believe that it is already an optional property on the stream, as in https://github.com/jordan-glitch/airbyte/blob/jordan-glitch-patch-1/airbyte-cdk/python/airbyte_cdk/models/airbyte_protocol.py#L199. All I would like is for this to be passed through to the catalog.

Reply:
Please let me know if I have misunderstood how the catalog is constructed.

Reply:
@jordan-glitch 

In the Airbyte protocol, a namespace is attached to a stream, not the whole source. For example, two streams in the same connector can come from two different namespaces (e.g: two tables in two different schemas in Postgres -- each schema is a separate namespace). 

So the first thing we need to do is declare a `@property` inside `stream.py`  in the `Stream` class that can be overridden by children of the `Stream` class. 

```
 @property
    def namespace(self) -> Optional[str]:
        """
header comment goes here
        """
        return None
```

The second thing is similar to what you have here inside `as_airbyte_stream()` but it should be wrapped in an `if` statement which checks if `self.namespace` contains e.g

```
if self.namespace and self.namespace != "":
  stream.namespace = self.namespace
```

Third we should add some unit tests on this functionality to ensure all the above works

Last, we should create a follow-up github issue which ensures that the CDK takes namespace into consideration when syncing streams selected in the input catalog. I'm happy to create it if you take care of the changes above

Does that clarify?

Reply:
Yes, thank you. I have made the above changes.